commands/compile/compile.go

```diff
@@ -88,20 +89,11 @@ func Compile(ctx context.Context, req *rpc.CompileReq, outStream, errStream io.W
 	builderCtx.SketchLocation = sketch.FullPath
 
 	// FIXME: This will be redundant when arduino-builder will be part of the cli
-	if packagesDir, err := config.HardwareDirectories(); err == nil {
-		builderCtx.HardwareDirs = packagesDir
-	} else {
-		return nil, fmt.Errorf("cannot get hardware directories: %s", err)
-	}
-
-	if toolsDir, err := config.BundleToolsDirectories(); err == nil {
-		builderCtx.BuiltInToolsDirs = toolsDir
-	} else {
-		return nil, fmt.Errorf("cannot get bundled tools directories: %s", err)
-	}
+	builderCtx.HardwareDirs = configuration.HardwareDirectories()
+	builderCtx.BuiltInToolsDirs = configuration.BundleToolsDirectories()
 
 	builderCtx.OtherLibrariesDirs = paths.NewPathList()
-	builderCtx.OtherLibrariesDirs.Add(config.LibrariesDir())
+	builderCtx.OtherLibrariesDirs.Add(paths.New(viper.GetString("directories.Libraries")))
```

Comment:
Why not put `viper.GetString("directories.Libraries")` into an helper function? like as you did with `configuration.HardwareDirectories()` for example?

At this point I'd write an helper function for each key we put in viper, doing so will enforce a more strict compile-time checks.

Reply:
One of the reason to switch to Viper was having a generic k/v store that can arbitrarily grow without the need to change the API. I understand the issue of losing the check on the configuration keys at compile time (if you write `viper.GetString("directories.Libs")` the program will happily compile despite not working as expected) but I think the tradeoff is worth it. We can catch errors on configuration keys at unit test time.